src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -452,6 +948,42 @@ public void Remove(string name)
                 throw new NotImplementedException();
             }
         }
+
+        /// <summary>
+        /// Holds state that we want to pass between diagnostic source events
+        /// </summary>
+        private class RequestTrackingFeature
+        {
+            /// <summary>
+            /// Gets or sets a value indicating whether the pipeline using endpoint routing
+            /// </summary>
+            public bool IsUsingEndpointRouting { get; set; }
+
+            /// <summary>
+            /// Gets or sets a value indicating whether this is the first pipeline execution
+            /// </summary>
+            public bool IsFirstPipelineExecution { get; set; } = true;
+
+            /// <summary>
+            /// Gets or sets a value indicating the route as calculated by endpoint routing (if available)
+            /// </summary>
+            public string Route { get; set; }
+
+            /// <summary>
+            /// Gets or sets a value indicating the resource name as calculated by the endpoint routing(if available)
+            /// </summary>
+            public string ResourceName { get; set; }
+
+            /// <summary>
+            /// Gets or sets the HTTP method, as it requires normalization, so avoids repeatedly calculations
+            /// </summary>
+            public string HttpMethod { get; set; }
+
+            /// <summary>
+            /// Gets or Sets the original URL received by the pipeline
+            /// </summary>
+            public string Url { get; set; }
```

Comment:
What do you think about renaming this to `OriginalUrl`?

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/1289/commits/f1245d9aabb243613ed22a52a8b708d86d3c9cd2